internal/getmodules/oci_getter.go

```diff
@@ -7,11 +7,424 @@ package getmodules
+		// We can potentially get in here if the author writes a bizarre
+		// source address with an explicit getter separate from the
+		// scheme, like oci::https://example.com/ .
+		return nil, fmt.Errorf("oci source type only supports oci URL scheme")
+		// Most authors will hopefully write just oci:// as the scheme part
+		// and let go-getter automatically select the oci getter based on
+		// that, in which case the scheme will be set correctly.
+	}
+
+	// As usual, registryDomainName can optionally include a trailing
+	// port number separated by a colon despite the name. This is a
+	// standard naming quirk in OCI Distribution implementations,
+	// since most addresses do not use an explicit port number.
+	registryDomainName := url.Host
+
+	// The OpenTofu module installer has already stripped off any
+	// "subdir" part of the path before calling us, so we can assume
+	// that the entire path is intended to be the repository name,
+	// except that the leading slash should not be included.
+	repositoryName := strings.TrimPrefix(url.Path, "/")
+
+	// We'll borrow ORAS-Go's implementation to validate the address elements.
+	ref := &orasRegistry.Reference{
+		Registry:   registryDomainName,
+		Repository: repositoryName,
+	}
+	if err := ref.Validate(); err != nil {
+		// ORAS-Go's error messages already include an "invalid reference:"
+		// prefix, so we won't add anything new here.
+		return nil, err
+	}
+	return ref, nil
+}
+
+func (g *ociDistributionGetter) resolveManifestDescriptor(ctx context.Context, ref *orasRegistry.Reference, query url.Values, store OCIRepositoryStore) (desc ociv1.Descriptor, err error) {
+	var unsupportedArgs []string
+	var wantTag string
+	var wantDigest ociDigest.Digest
+	for name, values := range query {
+		if len(values) > 1 {
+			return ociv1.Descriptor{}, fmt.Errorf("too many %q arguments", name)
+		}
+		value := values[0]
+		switch name {
+		case "tag":
+			if value == "" {
+				return ociv1.Descriptor{}, fmt.Errorf("tag argument must not be empty")
+			}
+			tagRef := *ref           // shallow copy so we can modify the reference field
+			tagRef.Reference = value // We'll again borrow the ORAS-Go validation for this
+			if err := tagRef.ValidateReferenceAsTag(); err != nil {
+				return ociv1.Descriptor{}, err // message includes suitable context prefix already
+			}
+			wantTag = value
+		case "digest":
+			if value == "" {
+				return ociv1.Descriptor{}, fmt.Errorf("digest argument must not be empty")
+			}
+			d, err := ociDigest.Parse(value)
+			if err != nil {
+				return ociv1.Descriptor{}, fmt.Errorf("invalid digest: %s", err)
+			}
+			wantDigest = d
+		default:
+			unsupportedArgs = append(unsupportedArgs, name)
+		}
+	}
+	if len(unsupportedArgs) == 1 {
+		return ociv1.Descriptor{}, fmt.Errorf("unsupported argument %q", unsupportedArgs[0])
+	} else if len(unsupportedArgs) >= 2 {
+		return ociv1.Descriptor{}, fmt.Errorf("unsupported arguments: %s", strings.Join(unsupportedArgs, ", "))
+	}
```

Comment:
Is there any particular reason for having these two conditionals? Wouldn't be enough just a `>=1`. Is this just for the `argument` vs `arguments` detail?
If so, TBH, not sure that it worths doing so.
I don't feel strongly about this, but don't see a lot of added value either. 🤷‍♂️

Reply:
Ok. I was just curious. Don't feel strongly about it either.
If you want to leave it like this, no complain from my side.
Thanks for the extra context 👍